enumarable_methods.rb

```diff
@@ -0,0 +1,74 @@
+module Enumerable
+  def my_each
+    each do |i|
+      yield i
+      return to_enum unless block_given?
+    end
+  end
+
+  def my_each_with_index
+    (0...length).each do |i|
+      yield(self[i], i)
+    end
+  end
```

Comment:
- [ ] should have working `my_each_with_index` identical and returns the same thing as ruby's [`each_with_index`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-each_with_index). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each_with_index.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- You are not supposed to use the original ruby methods but to create your own. In this case, you can use a while loop that finishes when you reached the last element of the enumerable.
- Your method is not returning an enumerator when no block is given.